enhancements/ingress/global-options-enable-hsts.md

```diff
@@ -0,0 +1,252 @@
+HSTS provides user protection and is concerned with minimizing security threats based on network
+traffic eavesdropping and man-in-the-middle attacks.  Using a response header 
+called `Strict-Transport-Security`, the HTTP response informs clients that a website 
+can only be accessed via HTTPS.  
+
+Administrators who are tasked with route management and/or regulatory compliance 
+issues want an option to automatically and globally enable HSTS. They also want 
+to separately apply HSTS globally to worker nodes and infrastructure nodes (e.g. 
+Prometheus, Grafana, web console hosts), while retaining the ability to opt-out 
+of HSTS on a per-route basis. 
+
+### Goals
+
+This proposal allows administrators to:
+- Continue to enable HSTS per-route (retain the existing feature functionality)
+- Enable HSTS for all routes globally
+- Enable HSTS on a node-type basis (infra-only, worker-only, or both)
+    - (TBD - waiting for clarification on this request.  Routes aren't associated with node types.)
+- Enable HSTS for specific subdomains 
+    - (TBD - dependency on wildcard DNS record/certificate, or just the `includeSubDomains`)
+- Disable HSTS per-route when it is enabled globally
+
+### Non-Goals
+
+- HSTS cannot be applied to non-TLS routes, even if HSTS is requested for all routes globally.  As
+mentioned in [RFC 6797 Section 8.1](https://tools.ietf.org/html/rfc6797#section-8.1), if an HTTP 
+response is received over insecure transport, clients must ignore any present STS headers.
+- The current implementation has a `preload` directive for the `Strict-Transport-Security`
+header.  This is not an RFC 6797 directive and therefore its implementation may vary by
+browser.  It can be specified, but support may vary by implementation.
+
+## Proposal
+
+Each goal is addressed in the subsections below.
+
+#### Continue to enable HSTS per-route (retain the existing feature functionality)
+The `Strict-Transport-Security` response header is currently applied per-route via an 
+annotation called `haproxy.router.openshift.io/hsts_header`.  This annotation has a
+required directive `max-age`, and optional directives `includeSubDomains` and `preload`,
+with the following definitions:
+- `max-age`: delta time range in seconds during which the host is regarded as an HSTS host
+    - If set to 0, it negates the effect and the host is no longer regarded as an HSTS host
+    - `max-age` decrements over time
+- `includeSubDomains`: if present, applies to any subdomains of the host's domain name
+- `preload`: if present, tells the client to include this host in its host preload list so that 
+it never needs to do an initial load to get the HSTS header (note that this is not defined in RFC 6797 
+and is therefore client implementation-dependent)
+
+HSTS is currently implemented as a pattern in the HAProxy template and applied to `edge` and 
+`reencrypt` routes that match the `haproxy.router.openshift.io/hsts_header` annotation:
+````
```

Comment:
I believe you want to use triple backticks for code fencing (as opposed to quadruple backticks).

Reply:
Ah I guess this is true (just as long as the entire doc is consistent :grin: )